diff -up platform/system/core/adb/client/usb_linux.cpp.0006~ platform/system/core/adb/client/usb_linux.cpp
--- platform/system/core/adb/client/usb_linux.cpp.0006~	2017-11-29 19:11:44.000000000 +0100
+++ platform/system/core/adb/client/usb_linux.cpp	2018-09-09 23:06:56.699814619 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <unistd.h>
 
@@ -271,7 +272,7 @@ static void find_usb_device(const std::s
                             // Determine the device path
                         if (!fstat(fd, &st) && S_ISCHR(st.st_mode)) {
                             snprintf(pathbuf, sizeof(pathbuf), "/sys/dev/char/%d:%d",
-                                     major(st.st_rdev), minor(st.st_rdev));
+                                     gnu_dev_major(st.st_rdev), gnu_dev_minor(st.st_rdev));
                             ssize_t link_len = readlink(pathbuf, link, sizeof(link) - 1);
                             if (link_len > 0) {
                                 link[link_len] = '\0';
