--- platform/system/core/adb/sockets.cpp.omv~	2017-05-07 23:07:44.985433626 +0200
+++ platform/system/core/adb/sockets.cpp	2017-05-07 23:08:38.119990370 +0200
@@ -614,7 +614,7 @@ char* skip_host_serial(const char* servi
 
     for (const std::string& prefix : prefixes) {
         if (!strncmp(service, prefix.c_str(), prefix.length())) {
-            return strchr(service + prefix.length(), ':');
+            return const_cast<char*>(strchr(service + prefix.length(), ':'));
         }
     }
 
@@ -623,13 +623,13 @@ char* skip_host_serial(const char* servi
         service += 4;
     }
 
-    char* first_colon = strchr(service, ':');
+    const char* first_colon = strchr(service, ':');
     if (!first_colon) {
         // No colon in service string.
         return nullptr;
     }
 
-    char* serial_end = first_colon;
+    const char* serial_end = first_colon;
     if (isdigit(serial_end[1])) {
         serial_end++;
         while (*serial_end && isdigit(*serial_end)) {
@@ -640,7 +640,7 @@ char* skip_host_serial(const char* servi
             serial_end = first_colon;
         }
     }
-    return serial_end;
+    return const_cast<char*>(serial_end);
 }
 
 }  // namespace internal
