--- platform/system/core/base/errors_unix.cpp.0001~	2017-10-20 09:17:28.000000000 +0200
+++ platform/system/core/base/errors_unix.cpp	2017-12-07 16:16:31.454622318 +0100
@@ -17,6 +17,7 @@
 #include "android-base/errors.h"
 
 #include <errno.h>
+#include <cstring>
 
 namespace android {
 namespace base {
--- platform/system/core/base/file.cpp.0001~	2017-10-20 09:17:28.000000000 +0200
+++ platform/system/core/base/file.cpp	2017-12-07 16:16:31.454622318 +0100
@@ -26,6 +26,7 @@
 #include <memory>
 #include <mutex>
 #include <string>
+#include <cstring>
 #include <vector>
 
 #include "android-base/logging.h"
--- platform/system/core/base/logging.cpp.0001~	2017-10-20 09:17:28.000000000 +0200
+++ platform/system/core/base/logging.cpp	2017-12-07 16:16:31.454622318 +0100
@@ -40,6 +40,7 @@
 #include <mutex>
 #include <sstream>
 #include <string>
+#include <cstring>
 #include <utility>
 #include <vector>
 
--- platform/system/core/libsparse/sparse_read.cpp.omv~	2017-12-07 16:16:56.287730712 +0100
+++ platform/system/core/libsparse/sparse_read.cpp	2017-12-07 16:17:02.860759675 +0100
@@ -26,6 +26,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include <unistd.h>
 
 #include <sparse/sparse.h>
--- platform/system/core/adb/sysdeps/posix/network.cpp.omv~	2017-12-07 17:03:46.469767388 +0100
+++ platform/system/core/adb/sysdeps/posix/network.cpp	2017-12-07 17:03:53.391809477 +0100
@@ -21,6 +21,7 @@
 #include <sys/socket.h>
 
 #include <string>
+#include <cstring>
 
 #include "adb_unique_fd.h"
 
