--- platform/system/core/base/file.cpp.omv~	2017-05-07 23:16:34.056665144 +0200
+++ platform/system/core/base/file.cpp	2017-05-07 23:16:44.369762070 +0200
@@ -22,6 +22,7 @@
 #include <sys/types.h>
 
 #include <string>
+#include <cstring>
 
 #include "android-base/macros.h"  // For TEMP_FAILURE_RETRY on Darwin.
 #include "android-base/utf8.h"
--- platform/system/core/base/logging.cpp.omv~	2017-05-07 23:19:06.151082378 +0200
+++ platform/system/core/base/logging.cpp	2017-05-07 23:19:12.927144965 +0200
@@ -33,6 +33,7 @@
 #include <limits>
 #include <sstream>
 #include <string>
+#include <cstring>
 #include <utility>
 #include <vector>
 
--- platform/system/core/base/errors_unix.cpp.omv~	2017-05-07 23:19:50.160488132 +0200
+++ platform/system/core/base/errors_unix.cpp	2017-05-07 23:19:56.369545241 +0200
@@ -17,6 +17,7 @@
 #include "android-base/errors.h"
 
 #include <errno.h>
+#include <cstring>
 
 namespace android {
 namespace base {
